src/test.js

```diff
@@ -550,6 +550,13 @@ function test( testName, expected, callback, async ) {
 	if ( arguments.length === 2 ) {
 		callback = expected;
 		expected = null;
+	} else if ( arguments.length === 3 && typeof expected !== "function" ) {
+		if ( global.console && global.console.warn ) {
+			global.console.warn(
+				"'expected' argument of QUnit.test is deprecated. " +
+				"Call the expect() function explicitly instead."
```

Comment:
This is a great start for #861, but we need to make a better structure out of it.

QUnit 1.x.x still supports browsers and other environments without the `console` object, or not even the `console.warn`.

I believe we should make a better structure first before triggering the warnings, like prefixing all the deprecation messages with an all caps `DEPRECATED`, also the HTML Reporter might get this info from the logs and show it on the reporter messages as well.

Also, I would like to have an internal method for the warning messages instead of checking for `console.warn` everywhere.